e2store/src/era.rs

```diff
@@ -108,12 +112,17 @@ impl Era {
             }))
     }
 
-    fn get_block_slot_indexes(slot_index_block_entry: &SlotIndexBlockEntry) -> Vec<u64> {
-        if slot_index_block_entry.slot_index.indices.is_empty() {
-            return vec![];
-        }
-        // minus 8 because the first 8 bytes are the starting slot
-        let beginning_of_index_record = slot_index_block_entry.slot_index.indices[0] - 8;
+    fn get_block_slot_indexes(
+        file_length: usize,
+        slot_index_block_entry: &SlotIndexBlockEntry,
+    ) -> Vec<u64> {
+        // To calculate the beginning of the index record, we need to subtract:
+        // - 8192*8 + 3*8 = 65560 bytes for the slot index block entry
+        // - 4*8=32 bytes for the slot index state entry
+        // from the total file length.
+        // Then subtract the result from u64::MAX to get the starting slot of the index record then
+        // add 1.
+        let beginning_of_index_record = u64::MAX - (file_length as u64 - (65560 + 32)) + 1;
```

Comment:
The indices/offsets should really be `i64`. Combined with comment above, this would be:
```rust
let beginning_of_index_record = file_length - SlotIndexBlockEntry::SERIALIZED_SIZE - SlotIndexStateEntry::SERIALIZED_SIZE;
let beginning_of_file_offset = -(beginning_of_index_record as i64);
```

And later you would just check `if *offset != beginning_of_file_offset {`